pkg/operator/sync.go

```diff
@@ -321,15 +322,25 @@ func (optr *Operator) syncRequiredMachineConfigPools(config renderConfig) error
 	if err != nil {
 		return err
 	}
-
+	isPoolStatusConditionTrue := func(pool *mcfgv1.MachineConfigPool, conditionType mcfgv1.MachineConfigPoolConditionType) bool {
+		for _, condition := range pool.Status.Conditions {
+			if condition.Type == conditionType {
+				return condition.Status == corev1.ConditionTrue
+			}
+		}
+		return false
+	}
 	for _, pool := range pools {
 		if err := isMachineConfigPoolConfigurationValid(pool, version.Version.String(), optr.mcLister.Get); err != nil {
 			return fmt.Errorf("pool %s has not progressed to latest configuration: %v, retrying", pool.Name, err)
 		}
-		if pool.Generation <= pool.Status.ObservedGeneration && pool.Status.MachineCount == pool.Status.UpdatedMachineCount && pool.Status.UnavailableMachineCount == 0 {
+		degraded := isPoolStatusConditionTrue(pool, mcfgv1.MachineConfigPoolDegraded)
+		if pool.Generation <= pool.Status.ObservedGeneration &&
+			isPoolStatusConditionTrue(pool, mcfgv1.MachineConfigPoolUpdated) &&
+			!degraded {
 			continue
 		}
-		return fmt.Errorf("error pool %s is not ready, retrying. Status: (total: %d, updated: %d, unavailable: %d)", pool.Name, pool.Status.MachineCount, pool.Status.UpdatedMachineCount, pool.Status.UnavailableMachineCount)
+		return fmt.Errorf("error pool %s is not ready, retrying. Status: (total: %d, updated: %d, ready: %d, unavailable: %d, pool degraded: %v)", pool.Name, pool.Status.MachineCount, pool.Status.UpdatedMachineCount, pool.Status.ReadyMachineCount, pool.Status.UnavailableMachineCount, degraded)
```

Comment:
i think order should be ready/updated/unavailable bc ready is the final state we are looking for right?

Reply:
yup, changing thanks!

Reply:
tho maybe leading with pool degraded would be clearer?

so pool degraded, total, ready updated, unavailable?

Reply:
pool degraded, total, ready, updated, unavailable looks good 💪